packages/contracts/src/strategies/BaseStrategy.sol

```diff
@@ -168,8 +171,6 @@ abstract contract BaseStrategy is
             totalDebt,
             gasCost
         );
-
-        emit Harvested(profit, loss, debtPayment, outstandingDebt);
```

Comment:
Why was this moved? "Harvested" event won't be emitted if the healthcheck failed and transaction reverted

Reply:
The reason for this change is recursive logic, in the case of not changing the place of emit logic this recursive logic will emit the event from the shutting down strategy first and only after it emits the event from the very initial withdrawal with the loss which is not correct order and it is better to emit it only once by the end of this method.

I thought about passing additional parameters or using extra variables (extra cost) but all those tricks look bad. Do you have an idea how to solve this issue effectively?

Reply:
nah, maybe it's ok, if healthcheck failed, that means no haverst was performed, so let's leave it as is